table-api/src/main/java/io/deephaven/api/filter/FilterOr.java

```diff
@@ -41,6 +40,20 @@ public static FilterOr of(Iterable<? extends Filter> filters) {
     @Parameter
```

Comment:
I wanted this comment to be on `FilterOr.of()` above (both overloads) but github won't let me click there.

It's tempting to do a little eager simplification based on the length of `filters`.  As in

```
if (filters.length == 0) {
   return Filter.ofFalse();  // empty OR is false
}
if (filters.length == 1) {
   return filters[0];
}
return builder().addFilters(filters).build();
```

This also reduces the number of cases where your `checkSize` method below would have to throw.

Again this would change the return value to just `Filter`

Reply:
I think it makes more sense to do this one level "higher", at `io.deephaven.api.filter.Filter#or(java.util.Collection<? extends io.deephaven.api.filter.Filter>)`. Ditto for `and`.

Reply:
I've update `Filter#or` and `Filter#and` in this style.